probe/endpoint/resolver.go

```diff
@@ -0,0 +1,72 @@
+package endpoint
+
+import (
+	"net"
+	"time"
+
+	"github.com/bluele/gcache"
+)
+
+const (
+	rAddrCacheLen        = 500 // Default cache length
+	rAddrBacklog         = 1000
+	rAddrCacheExpiration = 30 * time.Minute
+)
+
+// ReverseResolver is a caching, reverse resolver
+type ReverseResolver struct {
```

Comment:
Since this is only ever used within the endpoint package, can it be unexported, i.e. `type reverseResolver struct`? (That'll mean you need a resolver_internal_test.go for unit tests, but that's fine.)